test/javadoc-extension-test.js

```diff
@@ -0,0 +1,277 @@
+    contents = Buffer.from(contents)
+    const entry = {
+      contents,
+      src: { component, version, module, family, relative, path: relative },
+      pub: { moduleRootPath: '' },
+    }
+    contentCatalog.files.push(entry)
+    return entry
+  }
+
+  const createContentCatalog = () => ({
+    files: [],
+    findBy (criteria) {
+      const criteriaEntries = Object.entries(criteria)
+      const accum = []
+      for (const candidate of this.files) {
+        const candidateSrc = candidate.src
+        if (criteriaEntries.every(([key, val]) => candidateSrc[key] === val)) accum.push(candidate)
+      }
+      return accum
+    },
+  })
+
+  const run = (input = [], opts = {}) => {
+    file.contents = Buffer.from(Array.isArray(input) ? input.join('\n') : input)
+    const context = { config, contentCatalog, file }
+    opts.extension_registry = ext.register(opts.extension_registry || Asciidoctor.Extensions.create(), context)
+    opts.sourcemap = true
+    if (opts.convert) {
+      delete opts.convert
+      return log(Asciidoctor.convert(input, opts))
+    }
+    return log(Asciidoctor.load(input, opts))
+  }
+
+  function log (data) {
+    console.log(data)
+    return data
+  }
+
+  beforeEach(() => {
+    config = {}
+    contentCatalog = createContentCatalog()
+    file = addFile({ relative: 'index.adoc' }, '= Index Page')
+  })
+
+  describe('bootstrap', () => {
+    it('should be able to require extension', () => {
+      expect(ext).to.be.instanceOf(Object)
+      expect(ext.register).to.be.instanceOf(Function)
+    })
+
+    it('should not register to bound extension registry if register function called with no arguments', () => {
+      try {
+        ext.register.call(Asciidoctor.Extensions)
+        const extGroups = Asciidoctor.Extensions.getGroups()
+        const extGroupKeys = Object.keys(extGroups)
+        expect(extGroupKeys).to.be.empty()
+      } finally {
+        Asciidoctor.Extensions.unregisterAll()
+      }
+    })
+
+    it('should not register extension group if context is undefined', () => {
+      const input = []
+      const opts = { extension_registry: ext.register(Asciidoctor.Extensions.create()) }
+      const extensions = Asciidoctor.load(input, opts).getExtensions()
+      expect(extensions).to.be.undefined()
+    })
+
+    it('should be able to call register function exported by extension', () => {
+      const extensions = run().getExtensions()
+      expect(extensions.getInlineMacros()).to.have.lengthOf(1)
+    })
+  })
+
+  describe('javadoc macro', () => {
+    it('should convert using sensible defaults', () => {
+      const input = heredoc`
+        = Page Title
+
+        javadoc:com.example.MyClass[]
+        `
+      const actual = run(input, { convert: true })
+      expect(actual).to.include(
+        '<code class="apiref-hint:com.example.MyClass"><a href="attachment$api/java/com/example/MyClass.html" class="apiref">MyClass</a></code>'
```

Comment:
I might be missing something, but value of `attachment$` is part of the `xref` specification but it should not be in the `href` attribute of the html.

Reply:
I think in the real Antora it won't be (but I should test that assumption). In these tests we don't have [this code](https://gitlab.com/antora/antora/-/blob/8be02825ed544e3333ecb69113fa832f9da7faf6/packages/asciidoc-loader/lib/converter/html5.js#L29) so the xrefs aren't really understood.

Reply:
Thanks. I think that it would be ideal that the tests are setup so that our assertions reflect the expected html output.

Reply:
I'll look to see if we can pull in the Antora code.

Reply:
I've forced pushed something that uses the AntoraLoader and mimics resolving URLs